lib/menu.ts

```diff
@@ -148,41 +187,42 @@ function findAncestorChild(ancestor: Element, elem: Element|null): Element|null
 /**
  * Implements a menu item which opens a submenu.
  */
-export function menuItemSubmenu(submenu: MenuCreateFunc, ...args: DomElementArg[]): MenuItem {
+export function menuItemSubmenu(submenu: MenuCreateFunc, ...args: DomElementArg[]): Element {
   const ctl: PopupControl<IMenuOptions> = PopupControl.create(null);
 
   const popupOptions: IMenuOptions = {
     placement: 'right-start',
     trigger: ['click'],
-    attach: null,
     modifiers: {preventOverflow: {padding: 10}},
     boundaries: 'viewport',
     controller: ctl,
     isSubMenu: true,
   };
 
-  return (isSelected: Observable<boolean>) =>
-    cssMenuItem(...args, cssMenuItem.cls('-sel', isSelected),
-      dom('div', '\u25B6'),     // A right-pointing triangle
+  return cssMenuItem(...args,
+    dom('div', '\u25B6'),     // A right-pointing triangle
 
-      dom.autoDispose(ctl),
-      menu(submenu, popupOptions),
+    dom.autoDispose(ctl),
 
-      // On mouseover, open the submenu. Add a delay to avoid it on transient mouseovers.
-      dom.on('mouseover', () => ctl.open({showDelay: 250})),
+    // Set the submenu to be attached as a child of this element rather than as a sibling.
+    (elem: Element) => menuElem(elem, submenu, Object.assign(popupOptions, {attach: elem})),
 
-      // On right-arrow, open the submenu immediately, and select the first item automatically.
-      onKeyDown({
-        ArrowRight: () => ctl.open({startIndex: 0}),
-        Enter: () => ctl.open({startIndex: 0}),
-      }),
+    // On mouseover, open the submenu. Add a delay to avoid it on transient mouseovers.
+    dom.on('mouseover', () => ctl.open({showDelay: 250})),
 
-      // When selection changes, close the popup; remember to dispose this subscription.
-      dom.autoDispose(subscribe((use) => use(isSelected) || ctl.close())),
+    // On right-arrow, open the submenu immediately, and select the first item automatically.
+    onKeyDown({
+      ArrowRight: () => ctl.open({startIndex: 0}),
+      Enter: () => ctl.open({startIndex: 0}),
+    }),
 
-      // Clicks that open a submenu should not cause parent menu to close.
-      dom.on('click', (ev) => { ev.stopPropagation(); }),
-    );
+    // When selection changes, close the popup.
+    (elem: Element) => dom.dataElem(elem, 'menuItemSelected',
```

Comment:
I am guessing this doesn't work `dom.data('menuItemSelected', ...)`? Should we fix something?

Reply:
I think I was doing something with the element and never changed it